nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/model/impl/AbstractProperty.java

```diff
@@ -379,16 +381,25 @@ protected boolean isDeprecated() {
         return deprecated.booleanValue();
     }
 
+    protected boolean isSecured() {
+        PropertyCharacteristicHandler propertyHandler = Framework.getService(PropertyCharacteristicHandler.class);
+        // remove prefix
+        String xPath = getXPath();
+        String path = xPath.substring(xPath.lastIndexOf(':') + 1);
+        return propertyHandler.isSecured(getSchema().getName(), path) && !ClientLoginModule.isCurrentAdministrator();
+    }
+
     @Override
     public <T> T getValue(Class<T> type) throws PropertyException {
         return convertTo(getValue(), type);
     }
 
     @Override
     public void setValue(Object value) throws PropertyException {
-        // 1. check the read only flag
-        if (isReadOnly()) {
-            throw new ReadOnlyPropertyException(getXPath());
+        // 1. check the read only flag or security flag
+        if (isReadOnly() || isSecured()) {
+            throw new ReadOnlyPropertyException(
+                    String.format("Cannot set the value of property: %s since it is readonly", getXPath()));
         }
```

Comment:
is it  needed to treat the secured case and read only as the same. the exception and the message is the same in the both cases (Read Only, Secured). What's if we create two different exception or different messages.

Reply:
I think it's better to not have a specific message for secured because in such case we don't want to tell you don't have right to do that, no?

Note: read only is not used in Nuxeo.